main.vnet.tf

```diff
@@ -0,0 +1,15 @@
+module "avm_res_network_virtualnetwork" {
```

Comment:
This looks like it enables the creation of a new vnet, do we want to support existing vnets as well?

Reply:
The `count = var.associated_vnet == null ? 1 : 0` should cover this - so if an existing vnet is passed in then a new vnet is not created, the associated_vnet is used instead

Reply:
This should also have a case for vnet existing, but subnet not

Reply:
Makes sense - have added a default subnet that gets made if one isn't provided

Reply:
If resource id is provided for an existing vnet, the assumption is currently that the provided subnet via var exists. There still doesn't seem to be a path for existing vnet and create the subnet provided in variables

Reply:
Created issue #7 to address this